pages/index.jsx

```diff
@@ -30,48 +32,52 @@ export default function HomePage() {
               <Stack.Item fill>
                 <TextContainer spacing="loose">
                   <Text as="h2" variant="headingMd">
-                    Nice work on building a Shopify app 🎉
+                    {t("HomePage.heading")}
                   </Text>
                   <p>
-                    Your app is ready to explore! It contains everything you
-                    need to get started including the{" "}
-                    <Link url="https://polaris.shopify.com/" external>
-                      Polaris design system
-                    </Link>
-                    ,{" "}
-                    <Link url="https://shopify.dev/api/admin-graphql" external>
-                      Shopify Admin API
-                    </Link>
-                    , and{" "}
-                    <Link
-                      url="https://shopify.dev/apps/tools/app-bridge"
-                      external
-                    >
-                      App Bridge
-                    </Link>{" "}
-                    UI library and components.
+                    <Trans i18nKey="HomePage.yourAppIsReadyToExplore">
```

Comment:
Sure! My apologies, I should have highlighted this in the PR description.
The [`<Trans>` component](https://react.i18next.com/latest/trans-component) provides the ability to translate complex React elements. In this case, the translation is linked to the key [`HomePage.yourAppIsReadyToExplore`](https://github.com/Shopify/shopify-frontend-template-react/pull/176/files#diff-c9218bed23f0aeb93d7e1a7d5f6e1345fae20f4b8692b27fb97fd0f37b3d26d7R8).

It allows for the entire string to be translated without splitting into separate strings, with the React elements getting replaced with numeric placeholders in the translation file.

Without it, you would need to split the string into something like the following:
```jsx
<p>
  {t("HomePage.yourAppIsReadyToExplore", 
    polarisLink: `<Link url="https://polaris.shopify.com/" external>{t("HomePage.polarisLink")}</Link>`,
    adminApiLink: `<Link url="https://shopify.dev/api/admin-graphql" external>{t("HomePage.adminApiLink")</Link>`,
    appBridgeLink: `<Link url="https://shopify.dev/apps/tools/app-bridge" external>{t("HomePage.appBridgeLink")}</Link>`
  )}
</p>
```

The translation file would look like this:
```json
  "yourAppIsReadyToExplore": "Your app is ready to explore! It contains everything you need to get started including the {{polarisLink}}, {{adminApiLink}}, and {{appBridgeLink}} UI library and components.",
  "polarisLink": "Polaris design system",
  "adminApiLink": "Shopify Admin API",
  "appBridgeLink": "App Bridge",
```

It's not necessary to use the `<Trans>` component; it can just be more convenient for strings that have a bunch of nested components. 

One downside is that the JSX file now contains placeholder copy that isn't actually being used; the actual English strings come from the translation file. These strings could get out of sync, and a developer might misunderstand that the strings in the JSX file are the "real" ones.

Reply:
I could go either way on this. 

The `<Trans>` component is relatively simple to implement, but it requires some learning about how to structure the keys in the translation file. We could mitigate that by including some information and links in the README where we would list the i18n libraries used.

The benefits of having it in the template is that it demonstrates the usage of the different options available.
The downside is that it could be confusing and add to cognitive load.

I feel like my opinion is biased because of my relatively deep experience with i18n frameworks, so I defer to you and on your team on what approach is best for app developers.

Reply:
IMO we should optimize for "simplest to understand" and let developers progressively enhance/opt-in to additional complexity. I agree that the `<Trans>` component is not particularly complex on its own, but it did make me pause for a second to ask myself why there are two different approaches and which one would be better for my imaginary app 😂

Reply:
So, I just added a commit that replaces the `<Trans>` component with `t()` translate calls, and I observed a few interesting things. (Note, I didn't remove the previously generated keys, for easier comparison within the PR)

For the `HomePage.readyToExplore` key, I added three interpolation variables and three additional translation keys. It's not necessarily less complex than the `<Trans>` component, and it's possibly more prone to typos, though tophatting could reveal any issues there.

I'm using the [i18n-ally VSCode extension](https://github.com/lokalise/i18n-ally) to machine translate the content into French using Google Translate. I found that the machine translation was _much_ worse than when it translated from one large string in `HomePage.yourAppIsReadyToExplore`, because it translated each string fragment in isolation.
  - It incorrectly capitalized the first word of `HomePage.polarisLinkText`
  - It translated `HomePage.appBridgeLinkText` when it should probably stay untranslated ("App Bridge" being a proper noun of our library)
  - It rearranged the text of `HomePage.readyToExplore` incorrectly, so now it reads roughly like "Your app is ready to explore! It contains everything you need to get started, including the UI library and components {{polarisLink}}, {{adminApiLink}}, and {{appBridgeLink}}."

These mis-translations could be a result of the extension not being sophisticated enough to provide accurate translations with interpolation variables, but I think this highlights a common problem for machine and human translation alike - how difficult it is to provide accurate translations for string fragments that are intended to be used inside other strings.

If we want to point developers towards i18n practices that will result in higher quality translations for their applications, then I think it might be worth the additional cognitive load to include the `<Trans>` components in the default app template, with explanations and links in the README.

Reply:
UPDATE: Once I tophatted this, I discovered that we _must_ use the `<Trans>` component when there are React/HTML nodes integrated into a cohesive sentence, as we have in the two examples in this file.

Tophatted:
<img width="631" alt="Screenshot 2023-03-29 at 9 21 19 AM" src="https://user-images.githubusercontent.com/1530553/228573462-d028e4a5-2cc5-4ae1-be7c-b1b8075e2e22.png">

However, I discovered a format that provides named variables for the components (instead of arbitrary-looking numbers), and which IMO is a great middle ground of being easy to understand and being good for translators. That is what you see now in [this commit](https://github.com/Shopify/shopify-frontend-template-react/pull/176/commits/f8631722f163539bb972b5a754f9986bf3d2461c).

Tophatted:
<img width="638" alt="Screenshot 2023-03-29 at 9 29 42 AM" src="https://user-images.githubusercontent.com/1530553/228573690-012e9cc9-5c1e-412a-ba6c-9019fabd8053.png">